src/Microsoft.TestPlatform.Common/Interfaces/Engine/ClientProtocol/IProxyDiscoveryManager.cs

```diff
@@ -16,6 +16,15 @@ public interface IProxyDiscoveryManager
     /// </summary>
     void Initialize(bool skipDefaultAdapters);
 
+    /// <summary>
+    /// Initializes test discovery. Create the test host, setup channel and initialize extensions.
+    /// </summary>
+    /// 
+    /// <param name="discoveryCriteria">Settings, parameters for the discovery request</param>
+    /// <param name="eventHandler">EventHandler for handling discovery events from Engine</param>
+    /// <param name="skipDefaultAdapters">Skip default adapters flag</param>
+    void InitializeDiscovery(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEventsHandler2 eventHandler, bool skipDefaultAdapters);
```

Comment:
I missed that there was already some `Initialize` method above. Do we really need to keep both? Isn't it one of these many IXXX<NUMBER> interface were we would say the new way of doing it contains extra parameters?

Reply:
Not really, looks like we do need them both. `Initialize` initializes the proxy and we cannot get rid of this method (even though it's pretty vanilla) since it's public and has been on the interface forever. We don't want to alter its behavior either for the same reason as before (i.e. we don't know if/how this method is being used by other people, although the chances of someone actually implementing their own proxy and using it are really low). `InitializeDiscovery` initializes the discovery operation and is actually doing some heavy lifting like pre-starting a testhost that is ready to discover tests.